core/chains/evm/gas/rollups/arbitrum_l1_oracle.go

```diff
@@ -0,0 +1,313 @@
+	gasCostMethod      string
+	l1GasCostMethodAbi abi.ABI
+
+	priceReader daPriceReader
+
+	chInitialised chan struct{}
+	chStop        services.StopChan
+	chDone        chan struct{}
+}
+
+const (
+	// ArbGasInfoAddress is the address of the "Precompiled contract that exists in every Arbitrum chain."
+	// https://github.com/OffchainLabs/nitro/blob/f7645453cfc77bf3e3644ea1ac031eff629df325/contracts/src/precompiles/ArbGasInfo.sol
+	ArbGasInfoAddress = "0x000000000000000000000000000000000000006C"
+	// ArbGasInfo_getL1BaseFeeEstimate is the a hex encoded call to:
+	// `function getL1BaseFeeEstimate() external view returns (uint256);`
+	ArbGasInfo_getL1BaseFeeEstimate = "getL1BaseFeeEstimate"
+	// NodeInterfaceAddress is the address of the precompiled contract that is only available through RPC
+	// https://github.com/OffchainLabs/nitro/blob/e815395d2e91fb17f4634cad72198f6de79c6e61/nodeInterface/NodeInterface.go#L37
+	ArbNodeInterfaceAddress = "0x00000000000000000000000000000000000000C8"
+	// ArbGasInfo_getPricesInArbGas is the a hex encoded call to:
+	// `function gasEstimateL1Component(address to, bool contractCreation, bytes calldata data) external payable returns (uint64 gasEstimateForL1, uint256 baseFee, uint256 l1BaseFeeEstimate);`
+	ArbNodeInterface_gasEstimateL1Component = "gasEstimateL1Component"
+	// ArbGasInfo_getPricesInArbGas is the a hex encoded call to:
+	// `function getPricesInArbGas() external view returns (uint256, uint256, uint256);`
+	ArbGasInfo_getPricesInArbGas = "02199f34"
+)
+
+func NewArbitrumL1GasOracle(lggr logger.Logger, ethClient l1OracleClient) ArbL1GasOracle {
+	return newArbitrumL1GasOracle(lggr, ethClient)
+}
+
+func newArbitrumL1GasOracle(lggr logger.Logger, ethClient l1OracleClient) ArbL1GasOracle {
+	var l1GasPriceAddress, gasPriceMethod, l1GasCostAddress, gasCostMethod string
+	var l1GasPriceMethodAbi, l1GasCostMethodAbi abi.ABI
+	var gasPriceErr, gasCostErr error
+
+	l1GasPriceAddress = ArbGasInfoAddress
+	gasPriceMethod = ArbGasInfo_getL1BaseFeeEstimate
+	l1GasPriceMethodAbi, gasPriceErr = abi.JSON(strings.NewReader(GetL1BaseFeeEstimateAbiString))
+	l1GasCostAddress = ArbNodeInterfaceAddress
+	gasCostMethod = ArbNodeInterface_gasEstimateL1Component
+	l1GasCostMethodAbi, gasCostErr = abi.JSON(strings.NewReader(GasEstimateL1ComponentAbiString))
+
+	if gasPriceErr != nil {
+		panic("Failed to parse L1 gas price method ABI for chain: arbitrum")
+	}
+	if gasCostErr != nil {
+		panic("Failed to parse L1 gas cost method ABI for chain: arbitrum")
+	}
+
+	return &arbitrumL1Oracle{
+		client:     ethClient,
+		pollPeriod: PollPeriod,
+		logger:     logger.Sugared(logger.Named(lggr, "L1GasOracle(arbitrum)")),
+		chainType:  "arbitrum",
+
+		l1GasPriceAddress:   l1GasPriceAddress,
+		gasPriceMethod:      gasPriceMethod,
+		l1GasPriceMethodAbi: l1GasPriceMethodAbi,
+		l1GasCostAddress:    l1GasCostAddress,
+		gasCostMethod:       gasCostMethod,
+		l1GasCostMethodAbi:  l1GasCostMethodAbi,
+
+		priceReader: nil,
```

Comment:
Since this is always nil, I think we can exclude priceReader logic in this l1Oracle. Think that was Optimism specific anyways.